### PR TITLE
Fix Safari blank page when clicking on modular pipelines

### PR DESCRIPTION
**Pull request by vladimir-mck**
## Description

Resolves [1462](https://github.com/kedro-org/kedro-viz/issues/1462)

## Development notes

This issue specifically happens on Safari when interacting with modular pipelines in the sidebar or flowchart. The problem seems to arise during edge redraws. For some reason Safari has several `undefined` values in the edges array so I filtered them out when creating new edges.

This is the error Safari throws:
Screenshot 2023-08-15 at 12 57 48

And this is the error when I manually catch it in a try / catch block
Screenshot 2023-08-15 at 12 57 59

## QA notes

- Checkout branch
- Build & deploy 
- Open in Safari and start clicking modular pipelines in the sidebar or flowchart

## Checklist

- [x] Read the [contributing](/CONTRIBUTING.md) guidelines
- [ ] Opened this PR as a 'Draft Pull Request' if it is work-in-progress
- [ ] Updated the documentation to reflect the code changes
- [x] Added new entries to the `RELEASE.md` file
- [ ] Added tests to cover my changes

<a href="https://gitpod.io/#https://github.com/kedro-org/kedro-viz/pull/1488"></a>

### CODE REVIEW

**Review by tynandebold — APPROVED** [review]
Works very well. I can't reproduce the error now. Thanks for this!

**Review by ravi-kumar-pilla — APPROVED** [review]
LGTM !! Awesome :)

File: RELEASE.md [hunk 1]
```diff
@@ -17,6 +18,7 @@ Please follow the established format:
 - Fix broken SVG/PNG exports in light theme. (#1463)
 - Fix dataset and global toolbar error with standalone React component (#1351)
 - Fix `ImportError` as kedro-datasets is now lazily loaded (#1481).
+- Fix the issue of encountering a blank page in Safari when interacting with the modular pipeline. (#1462)
```
Comment (ravi-kumar-pilla) [inline RELEASE.md:21]: - Fix the issue of encountering a blank page in Safari when interacting with the modular pipeline. (#1488)

Comment (ravi-kumar-pilla) [reply RELEASE.md:21]: The # number in the bracket should be the PR number

**Review by ravi-kumar-pilla — CHANGES_REQUESTED** [review]

**Pull request merged**